src/components/SwapTab.tsx

```
@@ -1148,24 +1441,162 @@ export function SwapTab({
+        isThrottledRef.current = true;
+
+        // This is important to prevent further fetches
+        setTimeout(() => {
+          // Do this in next tick to ensure state is updated
+          quoteKeyRef.current = "completed-swap";
+        }, 0);
       } catch (psbtError) {
         // Re-throw to be caught by the outer catch block
         throw psbtError;
       }
     } catch (error: unknown) {
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)